test/common/http/utility_test.cc

```diff
@@ -124,6 +124,18 @@ TEST(HttpUtility, TestParseCookie) {
   EXPECT_EQ(value, "abc123");
 }
 
+TEST(HttpUtility, TestParseCookieBadValues) {
+  TestHeaderMapImpl headers{{"cookie", "token1=abc123; = "},
+                            {"cookie", "token2=abc123;   "},
+                            {"cookie", "; token3=abc123;"},
+                            {"cookie", "=; token4=\"abc123\""}};
+
+  EXPECT_EQ(Utility::parseCookieValue(headers, "token1"), "abc123");
```

Comment:
I would also make sure that you are exercising the error handling case. I'm not sure you actually are here with the way splitting is happening and break/return, but dunno.